File: src/hackerrank/practice/MakingAnagrams.java
package hackerrank.practice;

import java.util.Arrays;

public class MakingAnagrams {


    // Complete the makeAnagram function below.
    static int makeAnagram(String a, String b) {

        int[] counts = new int[26];

        for (char c : a.toCharArray()) {
            counts[c - 'a']++;
        }

        for (char c : b.toCharArray()) {
            counts[c - 'a']--;
        }
        return Arrays.stream(counts).map(v -> Math.abs(v)).sum();
    }
}

File: src/hackerrank/practice/MakingAnagrams.kt
package hackerrank.practice

import kotlin.math.absoluteValue

// Complete the makingAnagrams function below.
fun makingAnagrams(s1: String, s2: String): Int {

    val charMap = mutableMapOf<Char, Int>()

    val strMap1 = s1.groupingBy { it }.eachCount()
    val strMap2 = s2.groupingBy { it }.eachCount()

    var c : Char = 'a'

    while (c <='z') {
        charMap[c] = (strMap1.getOrDefault(c, 0) - strMap2.getOrDefault(c, 0)).absoluteValue
        c++

    }
    return charMap.map { it.value }.sum()

}